server/src/main/java/org/elasticsearch/reservedstate/service/AbstractFileWatchingService.java

```diff
@@ -363,11 +362,11 @@ final WatchKey enableDirectoryWatcher(WatchKey previousKey, Path settingsDir) th
     // package private for testing
     void processSettingsAndNotifyListeners() throws InterruptedException {
         try {
-            processFileChanges().get();
+            processFileChanges();
             for (var listener : eventListeners) {
                 listener.watchedFileChanged();
             }
-        } catch (ExecutionException e) {
+        } catch (IOException | ExecutionException e) {
```

Comment:
Previously all Exceptions were caught in processFileChanges. I think we want that same top level Exception caught here. What that means though is you'll. need to have a separate catch block for InterruptedException (which would just be rethrown)

Reply:
I've updated to match the comment. What do you think?